datumaro/components/project.py

```diff
@@ -2,486 +2,2399 @@
+                prev_stages = [self.make_target_name(n, t.head.name)
+                    for n, t in self.items() if n != self.MAIN_TARGET]
+            else:
+                prev_stages = [self.make_target_name(t, self[t].head.name)
+                    for t in target.parents]
 
-        self._length = None
+            for stage in target.stages:
+                stage_name = self.make_target_name(target_name, stage['name'])
+                graph.add_node(stage_name, config=stage)
+                for prev_stage in prev_stages:
+                    graph.add_edge(prev_stage, stage_name)
+                prev_stages = [stage_name]
 
-    def iterate_own(self):
-        return self.select(lambda item: not item.path)
+        return pipeline
 
-    def __iter__(self):
-        for subset in self._subsets.values():
-            yield from subset
+    def make_pipeline(self, target) -> Pipeline:
+        if not target in self:
+            raise UnknownTargetError(target)
 
-    def get_subset(self, name):
-        return self._subsets[name]
+        # a subgraph with all the target dependencies
+        if '.' not in target:
+            target = self.make_target_name(target, self[target].head.name)
 
-    def subsets(self):
-        return self._subsets
+        return self._make_full_pipeline().get_slice(target)
 
-    def categories(self):
-        return self._categories
+class GitWrapper:
+    @staticmethod
+    def import_module():
+        import git
+        return git
 
-    def __len__(self):
-        return sum(len(s) for s in self._subsets.values())
-
-    def get(self, id, subset=None, path=None): # pylint: disable=arguments-differ
-        if path:
-            source = path[0]
-            return self._sources[source].get(id=id, subset=subset)
-        return self._subsets.get(subset, {}).get(id)
-
-    def put(self, item, id=None, subset=None, path=None):
-        if path is None:
-            path = item.path
-
-        if path:
-            source = path[0]
-            # TODO: reverse remapping
-            self._sources[source].put(item, id=id, subset=subset, path=path[1:])
-
-        if id is None:
-            id = item.id
-        if subset is None:
-            subset = item.subset
-
-        item = item.wrap(path=path)
-        if subset not in self._subsets:
-            self._subsets[subset] = self.Subset(self, subset)
-        self._subsets[subset].items[id] = item
-        self._length = None
-
-        return item
-
-    def save(self, save_dir=None, merge=False, recursive=True,
-            save_images=False):
-        if save_dir is None:
-            assert self.config.project_dir
-            save_dir = self.config.project_dir
-            project = self._project
+    try:
+        module = import_module.__func__()
+    except ImportError:
+        module = None
+
+    def _git_dir(self):
+        return osp.join(self._project_dir, '.git')
+
+    def __init__(self, project_dir, repo=None):
```

Comment:
It was introduced for tests https://github.com/openvinotoolkit/datumaro/blob/276e4df87dbcea1d18dbe3e0507b79d5f3d3b45e/tests/test_project.py#L323. In this PR it does not matter, because git sources are not available yet.